src/librustc/hir/lowering.rs

```diff
@@ -1477,8 +1480,11 @@ impl<'a> LoweringContext<'a> {
                     && !self.already_defined_lifetimes.contains(&name) {
                     self.already_defined_lifetimes.insert(name);
 
+                    let ids = self.context.next_id();
```

Comment:
Self-nit: this (same in a few other spots) would be more concise with struct pattern matching.